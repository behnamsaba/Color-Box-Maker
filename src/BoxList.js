import { useState } from "react";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";
import './BoxList.css';
import { v4 as uuidv4 } from 'uuid';


const BoxList = () => {
    const INITIAL_state = [
        {color: "blue",width:200,height:200,id:uuidv4()}
    ];

    const [square, setSquare] = useState(INITIAL_state);

    const addSquare = (newSquare) => {
        setSquare(square => ([...square, {...newSquare,id:uuidv4()}]))
        
    }

    const removeSquare = (id) => {
        console.log("?!");
        setSquare(square.filter((object) => object.id === id));

    }

    return(
        <div className="BoxList">
            < NewBoxForm addSquare={addSquare}/>
            {square.map((square,id) => <Box key={id} id={uuidv4()} color={square.color} width={square.width} height={square.height} removeSquare={removeSquare}/>)}
        </div>
        
    )

}

export default BoxList