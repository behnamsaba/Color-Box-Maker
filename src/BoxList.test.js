import { fireEvent, queryByAttribute, render } from '@testing-library/react';
import BoxList from './BoxList';

test('renders without crashing', () => {
  render(<BoxList />);
});

test("match snapshot", function() {
  const {asFragment} = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
})


describe("add item from form", function() {
    test("page is empty at first", function() {
        const {queryByAttribute,queryByText,getByLabelText} = render(<BoxList />);
        expect(queryByText('X')).not.toBeInTheDocument();
    })
    test("add items work", function() {
        const {queryByRole,queryByText,getByLabelText} = render(<BoxList />);
        const input = getByLabelText('Enter Color');
        const btn = queryByText('Add');
        fireEvent.change(input, {target : {value : "black"}});
        fireEvent.click(btn);
        expect(queryByText("X")).toBeInTheDocument();

    })
})

test('form get empty after sumit', function() {
    const { getByLabelText,queryByText } = render(<BoxList/>);
    const input = getByLabelText('Enter Color');
    const btn = queryByText('Add');
    fireEvent.change(input, {target : {value : "black"}});
    fireEvent.click(btn);
    expect(input).not.toHaveValue();


})