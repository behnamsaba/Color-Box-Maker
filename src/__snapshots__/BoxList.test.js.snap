// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <form>
      <label
        for="color"
      >
        Enter Color
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <label
        for="width"
        id="width"
        name="width"
        value=""
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
